### PR TITLE
Enhance the validation for alias when creating index template or index

### PR DESCRIPTION
**Pull request by gaobinlong**
Relates to #61009.

The main changes are:
1. Throw an error if there are some unknown fields such as `bool` or some tokens such as `[]`  in the alias parameter when creating index template or creating index.
2. Add some tests for the change.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/action/admin/indices/alias/Alias.java [hunk 1]
```diff
@@ -223,6 +223,15 @@ public static Alias fromXContent(XContentParser parser) throws IOException {
         while ((token = parser.nextToken()) != XContentParser.Token.END_OBJECT) {
             if (token == XContentParser.Token.FIELD_NAME) {
                 currentFieldName = parser.currentName();
+                // check if there are any unknown fields
+                if (currentFieldName.equals(FILTER.getPreferredName()) == false &&
```
Comment (probakowski) [inline server/src/main/java/org/elasticsearch/action/admin/indices/alias/Alias.java:227]: Maybe extract all known names to set once and then just check it with `knownFields.contains(currentFieldName) == false`

**Review by probakowski — COMMENTED** [review]
Changes look good, thanks @gaobinlong. I have just one suggestion

File: server/src/main/java/org/elasticsearch/action/admin/indices/alias/Alias.java (hunk 1 above)
Comment (gaobinlong) [reply server/src/main/java/org/elasticsearch/action/admin/indices/alias/Alias.java]: OK, I have changed the code yet.

**Review by probakowski — APPROVED** [review]
LGTM, thanks @gaobinlong for iterating on this! I'll get this merged and backported

**Comment by danhermann** [comment]
This had to be reverted (see https://github.com/elastic/elasticsearch/pull/71739) because of a number of intake test failures. Once those have been resolved, we can re-open this one.

**Pull request merged**